### PR TITLE
Stats overhaul to support AMQP

### PR DESCRIPTION
**Pull request by aecay**
Changes in this PR:

- Start splitting out the different CLI commands into `mite/cli/<name>.py`.  So far only stats is on the new system
- Mild refactoring of the stats classes in `stats.py`
- Add unit tests for the stats classes in `stats.py` and `web/prometheus.py`
- Make the stats modular.  The core stats are separated from the http and selenium ones.  (Eventually we will have amqp stats as well, but these are not upstreamed yet...)
- Docs for the above, including a little bit about writing custom stats.

### CODE REVIEW

File: docs/stats.rst [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+available.  In order to pull in additional statistics, you will need to
+use the ``with_stats`` decorator on your scenario function, passing it
+the name(s) of the modules with custom stats that you want to be
+included:
+
+.. code-block:: python
+
+   from mite.stats import with_stats
+
+   @with_stats("mite_selenium")
```
Comment (jb098) [inline docs/stats.rst]: Needs to be updated

File: mite/__main__.py [hunk 2]
```diff
@@ -9,7 +9,7 @@
     mite [options] duplicator [--message-socket=SOCKET] OUT_SOCKET...
     mite [options] collector [--collector-socket=SOCKET]
     mite [options] recorder [--recorder-socket=SOCKET]
-    mite [options] stats [--stats-in-socket=SOCKET] [--stats-out-socket=SOCKET]
+    mite [options] stats [--stats-in-socket=SOCKET] [--stats-out-socket=SOCKET] [--scenario-spec=SPEC]
```
Comment (jb098) [inline mite/__main__.py]: Update this

File: mite/__main__.py [hunk 3]
```diff
@@ -42,6 +42,7 @@
     --collector-socket=SOCKET         Socket [default: tcp://127.0.0.1:14303]
     --stats-in-socket=SOCKET          Socket [default: tcp://127.0.0.1:14304]
     --stats-out-socket=SOCKET         Socket [default: tcp://127.0.0.1:14305]
+    --scenario-spec=SPEC              Importable spec for the scenario function, used to add stats
```
Comment (jb098) [inline mite/__main__.py]: And this

File: docs/stats.rst (hunk 1 above)
Comment (aecay) [reply docs/stats.rst]: Done in 2e4e21e

File: mite/__main__.py (hunk 2 above)
Comment (aecay) [reply mite/__main__.py]: Done in 2e4e21e

File: mite/__main__.py (hunk 3 above)
Comment (aecay) [reply mite/__main__.py]: Done in 2e4e21e

**Pull request merged**